metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -293,11 +300,47 @@ private void open() throws MetaException {
     boolean useFramedTransport = conf.getBoolVar(ConfVars.METASTORE_USE_THRIFT_FRAMED_TRANSPORT);
     int clientSocketTimeout = conf.getIntVar(ConfVars.METASTORE_CLIENT_SOCKET_TIMEOUT);
 
+    SocketFactory socketFactory = SocketFactory.getDefault();
+    Socket socket = null;
+    try {
+      socket = socketFactory.createSocket();
+      socket.setKeepAlive(true);
+
+      String hiveHost = conf.getVar(ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST);
```

Comment:
ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST exists on HiveConf.java. If the value is null, then hiveHost has empty since that is String class.